### PR TITLE
feat(rpm): support %config(missingok)

### LINKED ISSUE #853: Ability to specify %config(missingok)
### Is your feature request related to a problem? Please describe.

It's great to have a `%config(noreplace)` when building RPMs. Now we stumbled over the need to specify `%config(missingok)`.

This is especially made for cases where the package manager is happy with the file being there or missing. Given files you only need within postInstall scripts, symlinks that vanish or others.

### Describe the solution you'd like

Would be great to simply be able to pass `type: config|missingok` to a "content".

### Describe alternatives you've considered

Writing spec files by hand and not using nfpm. Which is somewhat not a real alternative :innocent: 

### Search

- [X] I did search for other open and closed issues before opening this.

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

### Additional context

https://rpm-software-management.github.io/rpm/manual/spec.html#missingok-since-rpm--414-in-standalone-form

### PR DESCRIPTION
**Pull request by caarlos0**
closes #853

### CODE REVIEW

**Review by robsliwi — COMMENTED** [review]
Somehow dnf running on AlmaLinux 8.10 is still complaining with a warning, though.
`fav32.png` is a `config|missingok` and `grafana_icon.svg` is a `config|noreplace` in a little Toy example:

```
Running transaction
  Preparing        :                                                                                                    1/1
  Erasing          : grafana-re-configuration-0.0.1-1.x86_64                                                            1/1
warning: file /tmp/grafana_icon.svg: remove failed: No such file or directory
warning: file /tmp/fav32.png: remove failed: No such file or directory
                                                                                                                            
  Verifying        : grafana-re-configuration-0.0.1-1.x86_64                                                            1/1
                                                                                                                            
Removed:
  grafana-re-configuration-0.0.1-1.x86_64
```

I will try to reproduce the toy example with a classic specfile and see if dnf complains with the warning here, too. But what for sure is possible with the specfile is to combine `missingok` and `noreplace` in a `%config(missingok,noreplace)` directive.


Was pleasantly surprised that direnv was asking me to allow the Nix devShell after checking out the repo :+1:

**Comment by robsliwi** [comment]
Small follow-up:

> I will try to reproduce the toy example with a classic specfile and see if dnf complains with the warning here, too. 

I did with the following spec file:
```
[root@d3d1a05f3904 ~]# cat hello.spec
Name:       hello-world
Version:    1
Release:    1
Summary:    Lets see if we can make it missingok
License:    FIXME
                                                                                                        
%description
This is my first RPM package, which does nothing.
                                                                                                        
%build
cat > config.conf <<EOF
# some config
EOF
                                                                                                        
%install
mkdir -p %{buildroot}/tmp/
install -m 755 config.conf %{buildroot}/tmp/config.conf
                                                                                                        
%files
%config(missingok) /tmp/config.conf
                                                                                                        
%changelog
# let's skip this for now
```

And the removal of the build RPM with a missing/moved `/tmp/config.conf`:

```
[root@d3d1a05f3904 ~]# mv /tmp/config.conf /tmp/config.conf.moved
mv: overwrite '/tmp/config.conf.moved'? y
[root@d3d1a05f3904 ~]# dnf remove hello-world
Dependencies resolved.
========================================================================================================
 Package                    Architecture          Version            Repository                    Size
========================================================================================================
Removing:
 hello-world                x86_64                1-1                @@commandline                 14
                                                                                                        
Transaction Summary
========================================================================================================
Remove  1 Package
                                                                                                        
Freed space: 14
Is this ok [y/N]: y
Running transaction check
Transaction check succeeded.
Running transaction test
Transaction test succeeded.
Running transaction
  Preparing        :                                                                                1/1
  Erasing          : hello-world-1-1.x86_64                                                         1/1
  Verifying        : hello-world-1-1.x86_64                                                         1/1
                                                                                                        
Removed:
  hello-world-1-1.x86_64
                                                                                                        
Complete!
```

No warnings here. Is there some place where I could have a peak at the generated spec file before "building" the RPM in the source code?

**Comment by caarlos0** [comment]
we don't create a spec file, that's why you don't need to have rpm installed to use this.

that said, I double checked https://refspecs.linuxbase.org/LSB_3.1.1/LSB-Core-generic/LSB-Core-generic/pkgformat.html#AEN27560 and google/rpmpack (which we use) and everything seems correct...

maybe its a bug on rpmpack? not sure yet

**Review by robsliwi — APPROVED** [review]
Just went through the examples and verified the file flags and those were set correctly.

```
rpm -qp --qf '[%{FILENAMES} %{FILEFLAGS:fflags}\n]' example-0.0.1-1.x86_64.rpm
/tmp/fav32.png cm
/tmp/grafana_icon.svg cn
```

And dnf is happy with all missing files with the newly introduced `cn` file flag :package: :tada: 

Thanks a lot for the blazing fast implementation!

The [duplicated line](https://github.com/goreleaser/nfpm/pull/854/files#r1746972044) is fine?

File: files/files.go [hunk 1]
```diff
@@ -32,9 +32,14 @@ const (
 	// user of the package.
 	TypeConfig = "config"
 	// TypeConfigNoReplace is like TypeConfig with an added noreplace directive
-	// that is respected by RPM-based distributions. For all other packages it
-	// is handled exactly like TypeConfig.
+	// that is respected by RPM-based distributions.
+	// that is respected by RPM-based distributions.
```
Comment (caarlos0) [inline files/files.go:36]:
```suggestion
```

**Pull request merged**